Mu2eG4/src/constructCRV.cc

```diff
@@ -76,23 +83,216 @@ namespace mu2e
+          const CLHEP::Hep3Vector &centerInMu2e0=layer0.getPosition();
+          const CLHEP::Hep3Vector &centerInMu2e1=layer1.getPosition();
+          motherCenterInMu2e.emplace_back(0.5*(centerInMu2e0+centerInMu2e1));
+          std::vector<double> halfLengths;
+          halfLengths.resize(3);
+          for(int i=0; i<3; ++i)
+          {
+            halfLengths[i]=layer0.getHalfLengths()[i]+0.5*fabs(centerInMu2e0[i]-centerInMu2e1[i]);
+          }
+          motherHalfLengths.push_back(halfLengths);
+        }
+
+        //aluminum absorbers dimensions
+        if(l+1<module0.getLayers().size())
+        {
+          const CRSAbsorberLayer &aluminumSheet0 = module0.getAbsorberLayers().at(l);
+          const CRSAbsorberLayer &aluminumSheet1 = module1.getAbsorberLayers().at(l);
+          const CLHEP::Hep3Vector &centerInMu2e0=aluminumSheet0.getPosition();
+          const CLHEP::Hep3Vector &centerInMu2e1=aluminumSheet1.getPosition();
+          motherCenterInMu2e.emplace_back(0.5*(centerInMu2e0+centerInMu2e1));
+          std::vector<double> halfLengths;
+          halfLengths.resize(3);
+          for(int i=0; i<3; ++i)
+          {
+            halfLengths[i]=aluminumSheet0.getHalfLengths()[i]+0.5*fabs(centerInMu2e0[i]-centerInMu2e1[i]);
+          }
+          motherHalfLengths.push_back(halfLengths);
+        }
+
+        //thin aluminum cover sheet dimensions
+        if(l+1==module0.getLayers().size())
+        {
+          const CRSAluminumSheet &aluminumSheet0 = module0.getAluminumSheets().at(1);
+          const CRSAluminumSheet &aluminumSheet1 = module1.getAluminumSheets().at(1);
+          const CLHEP::Hep3Vector &centerInMu2e0=aluminumSheet0.getPosition();
+          const CLHEP::Hep3Vector &centerInMu2e1=aluminumSheet1.getPosition();
+          motherCenterInMu2e.emplace_back(0.5*(centerInMu2e0+centerInMu2e1));
+          std::vector<double> halfLengths;
+          halfLengths.resize(3);
+          for(int i=0; i<3; ++i)
+          {
+            halfLengths[i]=aluminumSheet0.getHalfLengths()[i]+0.5*fabs(centerInMu2e0[i]-centerInMu2e1[i]);
+          }
+          motherHalfLengths.push_back(halfLengths);
+        }
+      }//dimensions of all layers 
+
+      //construct the mother solid for the CRV sector
+      //by combining the solids of each layer using G4UnionSolids
+      G4VSolid *motherSolid=nullptr;
+      for(size_t l=0; l<motherCenterInMu2e.size(); ++l)
+      {
+        //solid of one l-th layers of the CRV sector
+        G4VSolid *motherSolidPart=new G4Box(Form("%s_%s_%lu","CRSmotherSolidPart",CRVsectorName.c_str(),l),
+                                            motherHalfLengths[l][0],
+                                            motherHalfLengths[l][1],
+                                            motherHalfLengths[l][2]);
+        //need an additional solid that overlaps the solids of two adjacent layers,
+        //because G4UnionSolids require fully overlapping solids
```

Comment:
I think
// //because G4UnionSolids require constituent solids to have a fully overlapping region
would be more clear

Reply:
I agree with Krzysztof - describing this well will be the most valuable to a future maintainer.

Reply:
A technique that I find useful is to put numbered notes at the top of the file; then the comments in the code can refer to them.  That way a long comment does not disrupt the layout of the code and you can write as long a comment as is needed to tell the story well.  This is also useful if you need to make the same comment in several places.  I don't know if that will help in this cause - just be aware it's an option.